lib/spack/spack/package.py

```diff
@@ -779,6 +779,15 @@ def cleanup():
                          "Manually remove this directory to fix:",
                          self.prefix)
 
+            if not (keep_prefix and keep_stage):
+                self.do_clean()
+            else:
+                tty.warn("Keeping stage in place despite error.",
+                         "Spack will refuse to uninstall dependencies of this package." +
+                         "Manually remove this directory to fix:",
+                         self.stage.path)
```

Comment:
Having the stage in place was causing Spack to refuse to uninstall dependencies, although I don't think it will anymore since uninstall relies on the database to enforce dependencies now. Should I remove this now since it may be vestigial?